src/components/file-browser/FilesMenu.ts

```diff
@@ -67,6 +80,24 @@ export const FilesMenu = (props: FilesMenuProps) => {
       onClick: () => setConfirmingDelete(true),
     }, [icon('trash'), ' Delete']),
 
+    // @ts-expect-error
+    creatingNewDirectory && h(NameModal, {
+      thing: 'Folder',
```

Comment:
Is there any validation we want for the new directory name? A couple I'm worried about are all whitespace and a name with slashes "xxx/yyy". Maybe we could limit the name to alphanumeric characters, underscores and dashes? At a minimum we should restrict to the cloud provider requirements.

Reply:
Yikes, there is not. Not even a requirement that you enter a name. 

Found this documentation on object names: https://cloud.google.com/storage/docs/objects#naming

And this validation for folder names we use on the upload page.
https://github.com/DataBiosphere/terra-ui/blob/582ce9c472145a338f84252022b114394ea8ecf7/src/pages/Upload.js#L391-L397

Reply:
Added some validation (consistent with the data uploader) in https://github.com/DataBiosphere/terra-ui/pull/3653/commits/920168888e4754117cc3565c5d1e63213c4e186f. I think I'll open a separate PR to have NameModal require a value be input in order to submit.

Reply:
Updated NameModal to require input in #3712.